### PR TITLE
[PNI] Improved <title> tags for SEO

### PR DESCRIPTION
**Pull request by robdivincenzo**
# Description

This PR provides PNI wagtail pages with a better default `<title>` value that includes the `page.title`. The PR also enables editors to override the default `<title>` value with an SEO title tag on wagtail pages. These changes enable PNI for better localization and gracefully handle the existing hard-coded routable pages `pageTitle` values.

Notes for each page:

Homepage
- Currently hardcoded <title>: *Privacy Not Included | Shop smart and safe | Mozilla Foundation
  - Not localized, only in English for all languages, duplicated across the whole site
- PR will set <title> to the SEO title tag first if it exists
  - Pre-deploy we'll need to edit the SEO Title to: *Privacy Not Included | Shop smart and safe | Mozilla Foundation
   - The SEO title tag gets picked up for translation by Pontoon
     - Currently SEO title tag is set to *Privacy Not Included, so it is partially localized already

Reviews Index Page
- Currently inherits <title> from the homepage. Same issues, same post-PR resolution.
- Since it's inherited the <title> will be automatically resolved with homepage steps above

Reviews Category Page
- Currently <title> is fully hardcoded and localized to CATEGORY | Privacy & Security Guide | Mozilla Foundation
- Unchanged by PR

Articles Index Page
- Currently <title> inherits from homepage, same issues
- PR will set <title> to the SEO title tag first if it exists, But /articles does not have a title (although it now can in the future)
- PR then sets <title> to a default {{ page.title }} | *Privacy Not Included
  - Post-deploy, the title will be "Articles | *Privacy Not Included" with the ability to customize as desired vie SEO title tag
  - The page.title and *Privacy Not Included are both configured for localization

Article Page
- Currently <title> inherits from homepage, same issues
- PR resolves these same way as Articles Index Page

About Page(s)
- Currently <title> is fully hardcoded uniquely per page
- PR leaves it unchanged, but adds | Mozilla Foundation to the end for more consistency with the site (and within 60 character best practice)

Link to sample test page:
Related PRs/issues: #11277 #10241

# To test
- Edit /privacynotincluded homepage
  - Go to Promote and edit SEO title tag to be "*Privacy Not Included | Shop smart and safe | Mozilla Foundation"
  - View the page and verify the page title is "*Privacy Not Included | Shop smart and safe | Mozilla Foundation"
- Review the "Review All" index page
  - Verify the title is "*Privacy Not Included | Shop smart and safe | Mozilla Foundation"
- Review a product category review page
  - Verify the title is "CATEGORY | Privacy & Security Guide | Mozilla Foundation" where CATEGORY is the name of the category
- Review the Articles index page (/articles)
  - Verify the title is "Articles | *Privacy Not Included"
  - Edit the Articles index page, go to Promote, and edit the SEO title tag to be "Articles Test Title"
  - View and verify the Articles index page title is now "Articles Test Title"
- Review an individual articles page
  - Verify the title is "PAGE TITLE | *Privacy Not Included" where PAGE TITLE is the article title
  - Edit the Article page, go to Promote, and edit the SEO title tag to be "Another Test Title"
  - View and verify the Article page title is now "Another Test title"
- Review the About page
  - Verify the title is "Why we made *Privacy Not Included | Mozilla Foundation"

### CODE REVIEW

File: network-api/networkapi/templates/pages/buyersguide/base.html [hunk 1]
```diff
@@ -6,8 +6,16 @@
...
-    {% if pageTitle %}{{ pageTitle }}
-    {% else %}{% blocktrans context "“*Privacy Not Included” can be localized. This is a reference to the “*batteries not included” mention on toys." %}*Privacy Not Included | Shop smart and safe{% endblocktrans %} | Mozilla Foundation{% endif %}
+    {# if routable page #}
+    {% if pageTitle %}
+        {{ pageTitle }}
+    {# if custom seo_title for a wagtail page #}
+    {% elif page.seo_title %}
+        {{ page.seo_title }}
+    {% else %}
+    {# default to page.title #}
```
Comment (jhonatan-lopes) [inline network-api/networkapi/templates/pages/buyersguide/base.html:16]: For consistency with the comments above, I would swap these two lines. Not a blocker though, just a comment (#nitpick)

**Review by jhonatan-lopes — APPROVED** [review]
Thanks @robdivincenzo, LGTM :+1:

File: network-api/networkapi/templates/pages/buyersguide/base.html (hunk 1 above)
Comment (robdivincenzo) [reply network-api/networkapi/templates/pages/buyersguide/base.html]: Good point, I'll swap that out real quick before merge.

**Comment by robdivincenzo** [comment]
Reverted due to missing site identifier

**Pull request merged**